block/src/Pos/Block/Worker.hs

```diff
@@ -219,21 +221,34 @@ recoveryTriggerWorkerImpl SendActions{..} = do
     -- to initialize).
     delay $ sec 3
 
-    repeatOnInterval $
-        -- This is different from recoveryGuard, because latter
-        -- also doesn't run computation if we don't know current slot.
-        -- The following needRecovery function only checks if we
-        -- have recovery variable in place now and tries to fill it
-        -- (by requesting tips) if it's empty.
-        whenM (needRecovery @ctx) $ do
+    repeatOnInterval $ do
+        doTrigger <- needTriggerRecovery
+        when doTrigger $ do
+            logInfo "Triggering recovery because we need it"
             triggerRecovery enqueueMsg
-            -- We don't want to ask for tips too frequently.
-            -- E.g. there may be a tip processing mistake so that we
-            -- never go into recovery even though we recieve
-            -- headers. Or it may happen that we will receive only
-            -- useless broken tips for some reason (attack?). This
-            -- will minimize risks and network load.
-            delay $ sec 20
+
+        -- Sometimes we want to trigger recovery just in case. Maybe
+        -- we're just 5 slots late, but nobody wants to send us
+        -- blocks. It may happen sometimes, because nobody actually
+        -- guarantees that node will get updates on time. So we
+        -- sometimes ask for tips even if we're in relatively safe
+        -- situation.
+        (d :: Double) <- liftIO $ randomRIO (0,1)
+        unless (doTrigger && d < 0.005) $ do
```

Comment:
The probability to do it is 99.5%, is it intentional? Why not just 1?
Also you can use pattern matching for `doTrigger` instead of `when` followed by `unless`, but it's minor.